flox-bash/lib/commands/publish.sh

```diff
@@ -355,112 +375,129 @@ function floxPublish() {
 			registry "$gitCloneRegistry" 1 set buildRepository "$buildRepository"
 			registry "$gitCloneRegistry" 1 set packageAttrPath "$packageAttrPath"
 			registry "$gitCloneRegistry" 1 set channelRepository "$channelRepository"
 			registry "$gitCloneRegistry" 1 set uploadTo "$uploadTo"
 			registry "$gitCloneRegistry" 1 set downloadFrom "$downloadFrom"
 		fi
 	else
-		echo '{{ Color "'$LIGHTPEACH256'" "'$DARKBLUE256'" "'$entirePublishCommand'" }}' | \
+		echo '{{ Color "'$LIGHTPEACH256'" "'$DARKBLUE256'" "'"$entirePublishCommand"'" }}' | \
 			$_gum format -t template 1>&2
 	fi
 
 	# Start by making sure we can clone the channel repository to
 	# which we want to publish.
-	if [ "$channelRepository" = "-" ]; then
+	if [[ "$channelRepository" = "-" ]]; then
 		gitClone="-"
-	elif [ -d "$channelRepository" ]; then
+	elif [[ -d "$channelRepository" ]]; then
 		gitClone="$channelRepository"
 	else
-		gitClone=$tmpdir
+		gitClone="$tmpdir"
 		warn "Cloning $channelRepository ..."
 		$invoke_gh repo clone "$channelRepository" "$gitClone"
 	fi
 
 	# Then build package.
 	warn "Building $packageAttrPath ..."
-	local outpaths
-	outpaths=$(floxBuild "${_nixArgs[@]}" --no-link --print-out-paths "$canonicalFlakeURL" "${buildArgs[@]}")
-	[ -n "$outpaths" ] || error "could not build $canonicalFlakeURL" < /dev/null
+	declare -a outpaths
+	mapfile -t outpaths < <(
+		floxBuild "${_nixArgs[@]}" --no-link --print-out-paths  \
+		          "$canonicalFlakeURL^"'*' "${buildArgs[@]}"
+    )
+	if [[ "${#outpaths[@]}" -le 0 ]]; then
+		error "could not build $canonicalFlakeURL" < /dev/null
+	fi
 
-	# TODO Make content addressable (remove "false" below).
-	local ca_out
-	if false ca_out="$($invoke_nix "${_nixArgs[@]}" store make-content-addressed $outpaths --json | $_jq '.rewrites[]')"; then
+	# TODO Make content addressable (uncomment "XXX" lines below).
+	if false; then # XXX
+	declare -a ca_out
+	mapfile -t ca_out < <(
+		$invoke_nix "${_nixArgs[@]}" store make-content-addressed             \
+		            "${outpaths[@]}" --json|$_jq '.rewrites[]'||echo 'ERROR'
+	)
+	if [[ "${#ca_out[@]}" -gt 0 ]] && [[ "${#ca_out[0]}" != 'ERROR' ]]; then
 		# Replace package outpaths with CA versions.
 		warn "Replacing with content-addressable package: $ca_out"
-		outpaths=$ca_out
+		outpaths=( "${ca_out[@]}" )
 	fi
+	fi # /XXX
 
 	# Sign the package outpaths (optional). Sign by default?
-	if [ -z "$keyFile" -a -f "$FLOX_CONFIG_HOME/secret-key" ]; then
+	if [[ -z "${keyFile:-}" ]] && [[ -f "$FLOX_CONFIG_HOME/secret-key" ]]; then
 		keyFile="$FLOX_CONFIG_HOME/secret-key"
 	fi
-	if [ -n "$keyFile" ]; then
-		if [ -f "$keyFile" ]; then
-			$invoke_nix "${_nixArgs[@]}" store sign -r --key-file "$keyFile" $outpaths
+	if [[ -n "${keyFile:-}" ]]; then
+		if [[ -f "$keyFile" ]]; then
+			$invoke_nix "${_nixArgs[@]}" store sign -r --key-file "$keyFile"  \
+				        "${outpaths[@]}"
 		else
 			error "could not read $keyFile: $!" < /dev/null
 		fi
```

Comment:
nix will already raise an error if the keyfile is absent or corrupt fwiw

Reply:
Agreed ... will let `nix` flag corruption errors and we'll flag the simple existence check.  :+1: